### PR TITLE
8328279: Convert java/awt/Cursor/CursorOverlappedPanelsTest test to main

### PR DESCRIPTION
**Pull request by azuev-java**
Converted to a main manual test.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8328279](https://bugs.openjdk.org/browse/JDK-8328279): Convert java/awt/Cursor/CursorOverlappedPanelsTest test to main (**Bug** - P4)

### Reviewers
 * [Harshitha Onkar](https://openjdk.org/census#honkar) (@honkar-jdk - **Reviewer**)
 * [Abhishek Kumar](https://openjdk.org/census#abhiscxk) (@kumarabhi006 - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18331/head:pull/18331` \
`$ git checkout pull/18331`

Update a local copy of the PR: \
`$ git checkout pull/18331` \
`$ git pull https://git.openjdk.org/jdk.git pull/18331/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18331`

View PR using the GUI difftool: \
`$ git pr show -t 18331`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18331.diff">https://git.openjdk.org/jdk/pull/18331.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18331#issuecomment-2002158707)

### CODE REVIEW

File: test/jdk/java/awt/Cursor/CursorOverlappedPanelsTest.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+import java.awt.Dimension;
+import java.awt.Frame;
+import java.awt.Point;
+import java.lang.reflect.InvocationTargetException;
+import javax.swing.BorderFactory;
+import javax.swing.JFrame;
+import javax.swing.JLayeredPane;
+import javax.swing.JPanel;
+
+/**
```
Comment (honkar-jdk) [inline test/jdk/java/awt/Cursor/CursorOverlappedPanelsTest.java:34]:
```suggestion
/*
```

File: test/jdk/java/awt/Cursor/CursorOverlappedPanelsTest.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        JLayeredPane layeredPane = new JLayeredPane();
+        layeredPane.setPreferredSize(new Dimension(400, 400));
+        JPanel enabledPanel = createPanel(new Point(10, 10), true);
+        JPanel disabledPanel = createPanel(new Point(100, 100), false);
+        layeredPane.add(disabledPanel, JLayeredPane.PALETTE_LAYER);
+        layeredPane.add(enabledPanel, JLayeredPane.DEFAULT_LAYER);
+
+        frame.getContentPane().add(layeredPane);
+        frame.pack();
+        frame.setVisible(true);
```
Comment (honkar-jdk) [inline test/jdk/java/awt/Cursor/CursorOverlappedPanelsTest.java:56]: frame.setVisible(true) can be removed, this avoid the test pane to appear initially at a different location and then switch the final location, additionally PassFailJFrame handles it.

File: test/jdk/java/awt/Cursor/CursorOverlappedPanelsTest.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        PassFailJFrame.builder()
+                .title("Overlapping Panels Cursor Test Instructions")
+                .instructions(instructions)
+                .rows(4)
+                .columns(30)
+                .testUI(CursorOverlappedPanelsTest::initialize)
+                .build()
+                .awaitAndCheck();
+    }
+}
```
Comment (honkar-jdk) [inline test/jdk/java/awt/Cursor/CursorOverlappedPanelsTest.java:91]: EOF newline missing.

File: test/jdk/java/awt/Cursor/CursorOverlappedPanelsTest.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+            1) Move the mouse cursor into the area
+               of Enabled and Disabled panels intersection;
+            2) Check that the crosshair cursor is displayed.
+            If so, press PASS, otherwise press FAIL.
+            """;
+
+        PassFailJFrame.builder()
+                .title("Overlapping Panels Cursor Test Instructions")
+                .instructions(instructions)
+                .rows(4)
```
Comment (honkar-jdk) [inline test/jdk/java/awt/Cursor/CursorOverlappedPanelsTest.java:85]: May be 5-6 rows looks better for the instruction length.
Using  `.rows((int) instructions.lines().count() + 1)` avoids trying out different row count numbers :)

Comment (azuev-java) [reply test/jdk/java/awt/Cursor/CursorOverlappedPanelsTest.java]: Ok.

File: test/jdk/java/awt/Cursor/CursorOverlappedPanelsTest.java (hunk 2 above)
Comment (azuev-java) [reply test/jdk/java/awt/Cursor/CursorOverlappedPanelsTest.java]: Fixed.

File: test/jdk/java/awt/Cursor/CursorOverlappedPanelsTest.java (hunk 3 above)
Comment (azuev-java) [reply test/jdk/java/awt/Cursor/CursorOverlappedPanelsTest.java]: Added new line

File: test/jdk/java/awt/Cursor/CursorOverlappedPanelsTest.java (hunk 1 above)
Comment (azuev-java) [reply test/jdk/java/awt/Cursor/CursorOverlappedPanelsTest.java]: Done.

File: test/jdk/java/awt/Cursor/CursorOverlappedPanelsTest.java [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+ * @bug 8007155
+ * @summary [macosx] Disabled panel takes mouse input in JLayeredPane
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main/manual CursorOverlappedPanelsTest
+ */
+public class CursorOverlappedPanelsTest extends Frame {
+    public static JFrame initialize() {
+        final JFrame frame = new JFrame("Overlapping Panels Cursor Test");
+        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Cursor/CursorOverlappedPanelsTest.java:45]: frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE); redundant, can be removed.

File: test/jdk/java/awt/Cursor/CursorOverlappedPanelsTest.java [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+import javax.swing.JPanel;
+
+/*
+ * @test
+ * @bug 8007155
+ * @summary [macosx] Disabled panel takes mouse input in JLayeredPane
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main/manual CursorOverlappedPanelsTest
+ */
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Cursor/CursorOverlappedPanelsTest.java:41]: May add a blank line after this.

File: test/jdk/java/awt/Cursor/CursorOverlappedPanelsTest.java [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+import java.lang.reflect.InvocationTargetException;
+import javax.swing.BorderFactory;
+import javax.swing.JFrame;
+import javax.swing.JLayeredPane;
+import javax.swing.JPanel;
+
+/*
+ * @test
+ * @bug 8007155
+ * @summary [macosx] Disabled panel takes mouse input in JLayeredPane
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Cursor/CursorOverlappedPanelsTest.java:37]: Is the test exclusive for MacOS?

Comment (azuev-java) [reply test/jdk/java/awt/Cursor/CursorOverlappedPanelsTest.java:37]: No, the bug was found on MacOS but the test is valid for all operating systems.

File: test/jdk/java/awt/Cursor/CursorOverlappedPanelsTest.java (hunk 6 above)
Comment (azuev-java) [reply test/jdk/java/awt/Cursor/CursorOverlappedPanelsTest.java:41]: Ok

File: test/jdk/java/awt/Cursor/CursorOverlappedPanelsTest.java (hunk 5 above)
Comment (azuev-java) [reply test/jdk/java/awt/Cursor/CursorOverlappedPanelsTest.java:45]: Ok

**Review by honkar-jdk — APPROVED** [review]
LGTM

**Review by kumarabhi006 — APPROVED** [review]

**Pull request closed without merging**